### PR TITLE
Implement min reviewer branch policy + build validation + fix docs links

### PR DESCRIPTION
**Pull request by nmiodice**
## All Submissions:
-------------------------------------
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open PRs for the same update/change?

## What is the current behavior?
-------------------------------------
This change adds two main components:
 - Base code that enables configuration of all build policy types
 - Concrete implementation of `min reviewer` build policy
 - Concrete implementation of `build validation` build policy

It also fixes some missing documentation links for resources.

There will be a follow up story to implement other types of build policies, but they will not be included in this PR.

Closes:
 - https://github.com/microsoft/terraform-provider-azuredevops/issues/182
 - https://github.com/microsoft/terraform-provider-azuredevops/issues/184
 - https://github.com/microsoft/terraform-provider-azuredevops/issues/183

## What is the new behavior?
-------------------------------------
It is now possible to configure min reviewer branch policies like so:
```hcl
resource "azuredevops_branch_policy_min_reviewers" "p" {
  project_id = azuredevops_project.p.id

  enabled  = true
  blocking = true

  settings {
    reviewer_count     = 2
    submitter_can_vote = false

    scope {
      repository_id  = azuredevops_git_repository.r.id
      repository_ref = azuredevops_git_repository.r.default_branch
      match_type     = "Exact"
    }

    scope {
      repository_id  = azuredevops_git_repository.r.id
      repository_ref = "refs/heads/releases"
      match_type     = "Prefix"
    }
  }
}
```

or for a build validation:

```hcl
resource "azuredevops_branch_policy_build_validation" "p" {
  project_id = azuredevops_project.p.id

  enabled  = true
  blocking = true

  settings {
    display_name        = "Don't break the build!"
    build_definition_id = azuredevops_build_definition.b.id
    valid_duration      = 720

    scope {
      repository_id  = azuredevops_git_repository.r.id
      repository_ref = azuredevops_git_repository.r.default_branch
      match_type     = "Exact"
    }

    scope {
      repository_id  = azuredevops_git_repository.r.id
      repository_ref = "refs/heads/releases"
      match_type     = "Prefix"
    }
  }
}
```

## Does this introduce a breaking change?
-------------------------------------
- [NO]

## Other information
-------------------------------------
This change also updates the version of the `azure-devops-go-api` dependency, which includes a fix that enables the development of this feature.

### CODE REVIEW

**Comment by xuzhang3** [comment]
@nmiodice 
I try run this feature and get this error:
```
=== RUN   TestAccAzureDevOpsBranchPolicy_CreateAndUpdate
    TestAccAzureDevOpsBranchPolicy_CreateAndUpdate: testing.go:654: Step 0 error: errors during apply:

        Error: Error creating policy in Azure DevOps: parsing time ""2020-05-19T10:22:13.7774101"" as ""2006-01-02T15:04:05Z07:00"": cannot parse """ a
s "Z07:00"

```

**Comment by nmiodice** [comment]
@xuzhang3 that should have been fixed with the changes to `vendor` via the fix for https://github.com/microsoft/azure-devops-go-api/issues/59. I'll see if I can reproduce, but I consistently get passing `apply` both in acceptance test + local deployments

File: azuredevops/resource_branchpolicy_acceptance_test.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
+package azuredevops
```
Comment (xuzhang3) [inline azuredevops/resource_branchpolicy_acceptance_test.go:1]:
```suggestion
// +build all resource_branchpolicy_acceptance_test

package azuredevops
```

File: azuredevops/crud/branchpolicy/crud_branch_policy.go [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+}
+
+func flattenSettings(d *schema.ResourceData, policyConfig *policy.PolicyConfiguration) ([]interface{}, error) {
+	policySettings := commonPolicySettings{}
+	policyAsJSON, err := json.Marshal(policyConfig.Settings)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to marshal policy settings into JSON: %+v", err)
+	}
+
+	json.Unmarshal(policyAsJSON, &policySettings)
```
Comment (xuzhang3) [inline azuredevops/crud/branchpolicy/crud_branch_policy.go:106]:
```suggestion
        _ = json.Unmarshal(policyAsJSON, &policySettings)
	json.Unmarshal(policyAsJSON, &policySettings)
```

**Comment by xuzhang3** [comment]
> @xuzhang3 that should have been fixed with the changes to `vendor` via the fix for [microsoft/azure-devops-go-api#59](https://github.com/microsoft/azure-devops-go-api/issues/59). I'll see if I can reproduce, but I consistently get passing `apply` both in acceptance test + local deployments
> 
> @xuzhang3 can you try again? I suggest running `go mod vendor` to be sure you have the latest SDK, though I don't know why you'd _need_ to do that.

Update dependencies and refresh the `vendor` folder, it works for me.

**Comment by EliiseS** [comment]
```bash
Running tool: /usr/local/go/bin/go test -timeout 30s github.com/microsoft/terraform-provider-azuredevops/azuredevops -run ^(TestAccAzureDevOpsBranchPolicy_CreateAndUpdate)$ -v -tags=all

=== RUN   TestAccAzureDevOpsBranchPolicy_CreateAndUpdate
--- PASS: TestAccAzureDevOpsBranchPolicy_CreateAndUpdate (19.23s)
PASS
ok  	github.com/microsoft/terraform-provider-azuredevops/azuredevops	19.243s
```

Running the test works for me now too

File: azuredevops/resource_branchpolicy_acceptance_test.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+}
+
+type hclOptions struct {
+	projectName            string
+	repoName               string
+	minReviewerOptions     minReviewPolicyOpts
+	buildValidationOptions buildValidationPolicyOpts
+}
+
+func getHCL(opts hclOptions) string {
```
[the remark this answers is not in the document]
Comment (nmiodice) [reply azuredevops/resource_branchpolicy_acceptance_test.go:105]: HCL is *Hashicorp Configuration Language* - and is a fairly standard term IMO so I'll keep it unless you feel strongly. https://www.terraform.io/docs/configuration/syntax.html

File: azuredevops/resource_branchpolicy_acceptance_test.go [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+}
+
+type buildValidationPolicyOpts struct {
+	enabled       bool
+	blocking      bool
+	displayName   string
+	validDuration int
+}
+
+type hclOptions struct {
```
[the remark this answers is not in the document]
Comment (nmiodice) [reply azuredevops/resource_branchpolicy_acceptance_test.go:98]: See above

**Pull request merged**